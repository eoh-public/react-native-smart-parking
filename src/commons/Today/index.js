import React from 'react';
import { View, StyleSheet } from 'react-native';
import moment from 'moment';
import t from 'i18n';
import i18n from 'i18n-js';

import { Colors } from '../../configs';
import Text from '../../commons/Text';

const TodayComponent = ({ style }) => {
  const getCurrentDate = () => {
    const currentLanguage = i18n.currentLocale();
    moment.locale(currentLanguage);
    let format = 'MMM DD';
    if (currentLanguage === 'vi') {
      format = 'Do MMMM';
    }
    return t('Today') + ', ' + moment().format(format);
  };

  return (
    <View style={style}>
      <Text semibold style={styles.todayText}>
        {getCurrentDate()}
      </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  todayText: {
    color: Colors.Gray9,
    fontSize: 20,
  },
});

export default TodayComponent;
