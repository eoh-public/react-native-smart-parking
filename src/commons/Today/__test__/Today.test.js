import React from 'react';
import renderer, { act } from 'react-test-renderer';
import Today from '../index';

describe('Test Today', () => {
  let tree;
  test('create Today', () => {
    Date.now = jest.fn(() => new Date('2021-01-24T12:00:00.000Z'));
    act(() => {
      tree = renderer.create(<Today />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
