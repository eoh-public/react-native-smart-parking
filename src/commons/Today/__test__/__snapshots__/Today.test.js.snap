// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Today create Today 1`] = `
<View>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "SFProDisplay-Semibold",
          "fontSize": undefined,
          "lineHeight": undefined,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "color": "#262626",
          "fontSize": 20,
        },
      ]
    }
  >
    HÃ´m nay, 24th January
  </Text>
</View>
`;
